stub/include/neurala/image/views/ImageView.h

```diff
@@ -110,79 +136,67 @@ class ImageView
 	/// @copydoc ImageMetadata::datatype() const
 	constexpr EDatatype datatype() const noexcept { return m_metadata.datatype(); }
 
-	/// @copydoc ImageMetadata::sizeBytes() const
-	constexpr size_type sizeBytes() const noexcept { return m_metadata.sizeBytes(); }
+	/// @brief Returns the type of view strategy used.
+	virtual std::string viewStrategy() const noexcept { return "identity"; }
 
-	/**
-	 * @brief Returns the metadata associated with this view.
-	 */
-	constexpr const ImageMetadata& metadata() const noexcept { return m_metadata; }
+	/// @brief Returns the metadata associated with the viewed image.
+	constexpr const ImageMetadata& imageMetadata() const noexcept { return m_metadata; }
 
-	/**
-	 * @brief Returns the data associated with this view.
-	 */
-	constexpr const void* data() const noexcept { return m_data; }
+	/// @brief Returns the data associated with the viewed image.
+	constexpr const void* imageData() const noexcept { return m_data; }
 
-	/**
-	 * @brief Returns the data associated with this view as @p T.
-	 */
+	/// @brief Returns the data associated with the viewed image as @p T.
 	template<class T>
-	constexpr const T* dataAs() const noexcept
+	constexpr const T* imageDataAs() const noexcept
 	{
-		return static_cast<const T*>(data());
+		return static_cast<const T*>(imageData());
 	}
 
-	friend constexpr bool operator==(const ImageView& x, const ImageView& y) noexcept
+	virtual bool operator==(const ImageView& im) const noexcept
 	{
-		if (x.metadata() != y.metadata())
+		if (viewStrategy() != im.viewStrategy() || imageMetadata() != im.imageMetadata())
```

Comment:
An ROI is never considered equal to a full image. It could be done otherwise, by calling a second virtual function with reversed operands, but I'd rather stay away from that.

Reply:
Update: Enabled comparing views using different strategies.